Foundation/NSDate.swift

```diff
@@ -72,15 +72,51 @@ public class NSDate : NSObject, NSCopying, NSSecureCoding, NSCoding {
     public func encodeWithCoder(aCoder: NSCoder) {
         
     }
-    
+
+    /**
+     A string representation of the date object (read-only).
+
+     The representation is useful for debugging only.
+
+     There are a number of options to acquire a formatted string for a date
+     including: date formatters (see
+     [NSDateFormatter](//apple_ref/occ/cl/NSDateFormatter) and
+     [Data Formatting Guide](//apple_ref/doc/uid/10000029i)),
+     and the `NSDate` methods `descriptionWithLocale:`,
+     `dateWithCalendarFormat:timeZone:`, and
+     `descriptionWithCalendarFormat:timeZone:locale:`.
+     */
     public override var description: String {
         get {
-            return CFCopyDescription(_cfObject)._swiftObject
+            let dateFormatterRef = CFDateFormatterCreate(kCFAllocatorSystemDefault, nil, kCFDateFormatterFullStyle, kCFDateFormatterFullStyle)
+            let timeZone = CFTimeZoneCreateWithTimeIntervalFromGMT(kCFAllocatorSystemDefault, 0.0)
+            CFDateFormatterSetProperty(dateFormatterRef, kCFDateFormatterTimeZoneKey, timeZone)
+            CFDateFormatterSetFormat(dateFormatterRef, "uuuu-MM-dd HH:mm:ss '+0000'"._cfObject)
+
+            return CFDateFormatterCreateStringWithDate(kCFAllocatorSystemDefault, dateFormatterRef, _cfObject)._swiftObject
         }
     }
-    
+
+    /**
+     Returns a string representation of the receiver using the given locale.
+
+     - Parameter locale: An `NSLocale` object.
+
+       If you pass `nil`, `NSDate` formats the date in the same way as the
+       `description` property.
+
+     - Returns: A string representation of the receiver, using the given locale,
+       or if the locale argument is `nil`, in the international format
+       `YYYY-MM-DD HH:MM:SS ±HHMM`, where `±HHMM` represents the time zone
+       offset in hours and minutes from UTC (for example,
+       "2001-03-24 10:45:32 +0600")
+     */
     public func descriptionWithLocale(locale: AnyObject?) -> String {
-        return description
+        guard let aLocale = locale else { return description }
+        let dateFormatterRef = CFDateFormatterCreate(kCFAllocatorSystemDefault, (aLocale as! NSLocale)._cfObject, kCFDateFormatterFullStyle, kCFDateFormatterFullStyle)
+        CFDateFormatterSetProperty(dateFormatterRef, kCFDateFormatterTimeZoneKey, CFTimeZoneCopySystem())
```

Comment:
Actually here is what I got with current Darwin Foundation (I tested with Xcode 7.2/Swift 2.1.1.). It actually returns with locale's language and time zone. Can you please check it again? 
```swift
let d = NSDate(timeIntervalSince1970: 0)
d.descriptionWithLocale(nil) // 1970-01-01 00:00:00 +0000
d.descriptionWithLocale(NSLocale(localeIdentifier: "en_US")) // Thursday, January 1, 1970 at 9:00:00 AM Japan Standard Time
d.descriptionWithLocale(NSLocale(localeIdentifier: "ja_JP")) // 1970年1月1日木曜日 9時00分00秒 日本標準時
```